### PR TITLE
update psutil to 6, furo to 2024, markupsafe to <3

### LINKED ISSUE #240: Update to psutil 6
https://github.com/PolicyEngine/policyengine-core/blob/47f5bdf953d8c6776f9af1f3859864c12ebbf36f/setup.py#L22

Now 6.0.0: https://pypi.org/project/psutil/

### LINKED ISSUE #241: Update to furo 2024
https://github.com/PolicyEngine/policyengine-core/blob/47f5bdf953d8c6776f9af1f3859864c12ebbf36f/setup.py#L34

Currently 2024.8.6: https://pypi.org/project/furo/

### LINKED ISSUE #242: Uncap markupsafe
Let's go to `<3`

https://github.com/PolicyEngine/policyengine-core/blob/47f5bdf953d8c6776f9af1f3859864c12ebbf36f/setup.py#L35

Currently 2.1.5: https://pypi.org/project/MarkupSafe/

### PR DESCRIPTION
**Pull request by irisbian504**
fix #240
fix #241
fix #242

### CODE REVIEW

**Review by anth-volk — CHANGES_REQUESTED** [review]
Thanks for this @irisbian504! 

It looks like you committed the `ipython` version update off of a local `master` branch, then opened this one in order to add the later commits. Unfortunately, this breaks two parts of our CI/CD Actions. In order to fix this, could you do the following:
1. Run `git pull upstream master`, then `git push origin master` (or in GitHub, click the button to keep your local fork up-to-date)
2. Switch to your branch, then run `git rebase master`. You'll probably hit a couple merge conflicts, but the changelog should only contain your most recent changes, and the version number in `setup.py` should be the most recent one, 3.11.1.

**Comment by anth-volk** [comment]
I took the liberty of rebasing, per the instructions above, as I work through some backlogged PRs.

File: changelog_entry.yaml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+- bump: minor
```
Comment (MaxGhenis) [inline changelog_entry.yaml:1]:
```suggestion
- bump: patch
```

File: changelog_entry.yaml [hunk 2]
```diff
@@ -0,0 +1,6 @@
...
+  changes:
+    changed:
+    - update the psutil requirement to version 6
+    - update the furo requirment to <2025
+    - update the markupsafe requirment to <3
```
Comment (MaxGhenis) [inline changelog_entry.yaml:6]:
```suggestion
    - update the markupsafe requirment to <3

```

File: setup.py [hunk 3]
```diff
@@ -15,7 +15,7 @@
     "sortedcontainers<3",
     "numexpr<3",
     "dpath<3",
-    "psutil<6",
+    "psutil>=6,<7",
```
Comment (MaxGhenis) [inline setup.py:18]: should we be consistent about `>=lower,<bound` or just do `<bound` like below?

**Review by MaxGhenis — CHANGES_REQUESTED** [review]

File: changelog_entry.yaml (hunk 2 above)
Comment (anth-volk) [inline changelog_entry.yaml:6]:
```suggestion
    - update the markupsafe requirement to <3
```

**Review by anth-volk — CHANGES_REQUESTED** [review]

File: setup.py (hunk 3 above)
Comment (anth-volk) [reply setup.py:18]: I think it depends on the package itself, but I don't think consistency on this is a high priority in my opinion.

**Pull request merged**